viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
+# test_wave_01 - line 61 to line 81
+def add_to_watched(user_data,movie):
+    user_data['watched'].append(movie)
+    return user_data
 
+# test_wave_01 line 104 to line 121  
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+# test_wave_01
+def watch_movie(user_data,title):
+    
+    watch_list = user_data['watchlist']
+    for index in range(len(watch_list)):
+        movie = watch_list[index]
+        if movie['title']==title:
+            watch_list.pop(index)
+            user_data['watched'].append(movie)
+    return user_data
 
-# -----------------------------------------
-# ------------- WAVE 3 --------------------
-# -----------------------------------------
+# -----------test_wave_02----------
+# line 5 to line 28 on test_wave02
+
+# user_data is a dict with they key=watch list with a dict of movies. 
+# movie has 3 keys: title, genre, rating
+    # movie_list=user_data['watched']
+def get_watched_avg_rating(user_data):
+    print(user_data)
+    count=0
+    score_sum=0
+    
+    if user_data['watched']==[]:
+        average_rating=0.0
+    else:
+        for movie in user_data['watched']:
+            score_sum+= movie['rating']
+            count+=1
+        average_rating=score_sum/count
+    return average_rating
+
+# testwave 02 - line 29 on   
+def get_most_watched_genre(user_data):
+    movie_list=user_data['watched']
+    genre_list=[]
+    genre_dict={}
+    count=0
+    if movie_list==[]:
+        return None
+    else:
+        for movie in movie_list:
+            genre_list.append(movie['genre'])
+        print(f' the genre list is {genre_list}')            
+    for i in genre_list:
+        if i in genre_dict:
+            genre_dict[i]+=1
+        else:
+            genre_dict[i]=1
+    print(f' the genre dict is {genre_dict}')
+
+    most_often=0
+    movies_most_often=None
+    for genre in genre_dict:
+        if genre_dict[genre]>most_often:
+            most_often=genre_dict[genre]
+            movies_most_often=genre
+        print(f' most often is {movies_most_often}')
+        print(f' dict is {genre_dict}')
+    
+    return movies_most_often
+        
+#--------------test_wave_03-------------
+
+def get_unique_watched(user_data):
+    user_watched_list= user_data['watched']
+    friends_list=user_data['friends']
+    friends_movie_list=[]
+    title_set=set()
+    user_unique_movie_list=[]
+    
+    for friend in friends_list:
+        for movie in friend['watched']:
+            if movie['title'] not in title_set:
+                title_set.add(movie['title'])
+                friends_movie_list.append(movie)
+    
+    for movie in user_watched_list:
+        unique = True
+        for fmovie in friends_movie_list:
+            if movie['title'] == fmovie['title']:
+                unique=False
+                break      
+        if unique:
+                user_unique_movie_list.append(movie)
+    return user_unique_movie_list

```

Comment:
Hmm, there's a lot going on here. I don't think we need these flags at all. The for loop will work fine without them.

```suggestion
    for movie in user_watched_list:
        if movie['title'] not in friends_movie_list:
            user_unique_movie_list.append(movie)
    return user_unique_movie_list
```